src/babashka/fs.cljc

```diff
@@ -799,3 +799,45 @@
                                              new-path
                                              cp-opts))))
              (recur))))))))
+
+(defn- with-temp-dir-fn
+  [[sym options] body]
+  `(let [~sym (create-temp-dir ~options)]
+     (try
+       ~@body
+       (finally
+         (delete-tree ~sym)))))
+
+(defn ^:no-doc with-temp-dir-bindings
+  [bindings body]
+  {:pre [(vector? bindings) (even? (count bindings))]}
+  (cond
+    (not (seq bindings)) `(do ~@body)
+
+    (symbol? (bindings 0))
+    (with-temp-dir-fn (subvec bindings 0 2)
+      (if-let [more-bindings (not-empty (subvec bindings 2))]
+        [`(with-temp-dir
+            ~more-bindings
+            ~@body)]
+        body))
+
+    :else
+    (throw
+     (IllegalArgumentException.
+      "with-temp-dir only allows [symbol value] pairs in bindings"))))
+
+(defmacro with-temp-dir
+  "bindings [binding-name options ...]
+
+  Evaluate body with binding-name bound to a java.io.File path to a
```

Comment:
the binding-name is bound to a file. in babashka.fs this is either a nio Path or java io File. You should not rely on this detail but if you care specifically, use fs/file or fs/path to convert between the two. So I would just call this "file" without specifying the type.